src/meet-calendar.ts

```diff
@@ -0,0 +1,688 @@
+
+  /**
+   * Updates meetingId, if there is meetingId set it, if not generate it.
+   */
+  updateMeetingId() {
+    if (!this.isButtonPresent()) {
+      // there is no button present we will add it, so we will clean
+      // the state of the EventContainer, so we can update all values.
+      // this clears the states between creating/editing different events
+      // we add the button
+      this.reset();
+    }
+
+    let inviteText;
+    let ix = -1;
+
+    // checking location
+    if (this.location && this.location.text) {
+      inviteText = this.location.text;
+
+      if (inviteText) ix = inviteText.indexOf(BASE_URL);
+    }
+
+    // if nothing found let's check description
+    if (ix == -1) {
+      inviteText = this.description.value;
+
+      if (inviteText) ix = inviteText.indexOf(BASE_URL);
+    }
+
+    var url;
+    if (ix != -1 && (url = inviteText?.substring(ix)) && url.length > 0) {
+      let resMeetingId = url.substring(BASE_URL.length);
+
+      // there can be ',' after the meeting, normally added when adding
+      // physical rooms to the meeting
+
+      var regexp = /([a-zA-Z0-9-_]+).*/g;
+      var match = regexp.exec(resMeetingId);
+      if (match && match.length > 1) resMeetingId = match[1];
+
+      this.meetingId = resMeetingId;
+    } else {
+      this.meetingId = generateRoomWithoutSeparator();
+
+      if (this.scheduleAutoCreateMeeting) {
+        this.description.clickAddMeeting(false, this.location);
+        this.scheduleAutoCreateMeeting = false;
+      }
+    }
+  }
+
+  /**
+   * Adds the jitsi button in buttonContainer.
+   */
+  addJitsiButton() {
+    var container = this.buttonContainer;
+    if (!container) return;
+
+    var description = this.description;
+
+    container.addClass("button_container");
+    container.append(
+      '<div id="jitsi_button" ' +
+        'class="goog-inline-block jfk-button jfk-button-action ' +
+        'jfk-button-clear-outline">' +
+        '<a href="#" style="color: white"></a>' +
+        "</div>"
+    );
+    description.update(this.location);
+  }
+}
+
+/**
+ * Represents the location field.
+ */
+abstract class LocationWrapper {
+  /**
+   * The text in the location field.
+   * @abstract
+   */
+  abstract get text(): string | undefined;
+
+  /**
+   * Adds location info.
+   * @abstract
+   * @param text
+   */
+  abstract addLocationText(text: string): void;
+}
+
+/**
+ * Represents the description of the event.
+ */
+abstract class DescriptionWrapper {
+  protected event: EventContainer;
+
+  constructor(event: EventContainer) {
+    this.event = event;
+  }
+  /**
+   * Updates the description and location field is not already updated.
+   */
+  update(location: LocationWrapper) {
+    let isDescriptionUpdated = false;
+
+    // checks whether description was updated.
+    if (this.element !== undefined) {
+      let descriptionContainsURL =
+        this.value &&
+        this.value.length >= 1 &&
+        this.value.indexOf(BASE_URL) !== -1;
```

Comment:
should use `startsWith` otherwise `https://bad.com/#https://$BASE_URL` would pass this check

Reply:
In this specific case, that is the behaviour we want. The description is the large free text box within the calendar entry. This bit of code is checking whether the description has previously been edited to include any kind of `talk.brave.com` url anywhere within it - if so, the description is left alone. This code is not attempting to perform url validation.

If the user (or another extension) has entered `https://bad.com/#https://talk.brave.com` into the description box then it is correct that we should leave it alone. We do not extract the url nor do we open it or follow it.

Reply:
got it, thanks